src/hooks/useDebounce.js

```diff
@@ -0,0 +1,17 @@
+import {
+  useCallback,
+  useEffect
+} from 'preact/hooks';
+
+export function useDebounce(callback, debounceFn) {
+  const debouncedCallback = useCallback(debounceFn(callback), [ callback, debounceFn ]);
+
+  // cancel previous function when new is created
```

Comment:
This should be adjusted after we decided to go for flush instead of cancel.

Reply:
Or removed...

Reply:
I've updated the comment to reflect the intent.